src/services/azureFunctionsService.ts

```diff
@@ -75,6 +75,7 @@ export class AzureFunctionsService implements mssql.IAzureFunctionsService {
 		// because of an AF extension API issue, we have to get the newly created file by adding
 		// a watcher: https://github.com/microsoft/vscode-azurefunctions/issues/2908
 		const newFilePromise = azureFunctionUtils.waitForNewFunctionFile(projectFile);
+		const checkFunctionFile = waitForPromise();
```

Comment:
As discussed earlier we shouldn't start the timeout until AFTER we've gotten the function name from the user and called createFunction.